### PR TITLE
add 'topic type' verb

### PR DESCRIPTION
**Pull request by artivis**
Add the `type` verb which return the type of a given topic.
E.g.
```
$ ros2 topic type /rosout
rcl_interfaces/msg/Log
```

### CODE REVIEW

**Comment by artivis** [comment]
Came out of discussing https://github.com/ros2/ros2cli/pull/204.

File: ros2topic/ros2topic/verb/type.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+from ros2topic.verb import VerbExtension
+
+
+class TypeVerb(VerbExtension):
+    """Print a topic's type."""
+
+    def add_arguments(self, parser, cli_name):
+        arg = parser.add_argument(
+            'topic_name',
+            help="Name of the ROS topic to get info (e.g. '/chatter')")
```
Comment (dirk-thomas) [inline ros2topic/ros2topic/verb/type.py]: The help text should mention the type instead of "info".

File: ros2topic/ros2topic/verb/type.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            include_hidden_topics_key='include_hidden_topics')
+
+    def main(self, *, args):
+        with DirectNode(args) as node:
+            topic_names_and_types = get_topic_names_and_types(
+                node=node,
+                include_hidden_topics=args.include_hidden_topics)
+
+        for (topic_name, topic_types) in topic_names_and_types:
+            if args.topic_name == topic_name:
```
Comment (dirk-thomas) [inline ros2topic/ros2topic/verb/type.py]: If the topic name is not found the function should return a non-zero exit code to indicate that it failed to find the given topic.

File: ros2topic/ros2topic/verb/type.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+                print(topic_types[0])
```
Comment (dirk-thomas) [inline ros2topic/ros2topic/verb/type.py]: By definition there can't be two tuples with the same topic name. So ince the topic is found the loop can return early to improve performance.

File: ros2topic/ros2topic/verb/type.py (hunk 1 above)
Comment (artivis) [reply ros2topic/ros2topic/verb/type.py]: Fixed.

File: ros2topic/ros2topic/verb/type.py (hunk 2 above)
Comment (artivis) [reply ros2topic/ros2topic/verb/type.py]: Went with a plain `return 1`, is that what you meant ?

File: ros2topic/ros2topic/verb/type.py (hunk 3 above)
Comment (artivis) [reply ros2topic/ros2topic/verb/type.py]: Went with a plain `return 0`, in mirror of the "non-zero exit code".

**Comment by dirk-thomas** [comment]
Thanks for the patch and the quick iterations!

Linux CI: [Build Status](https://ci.ros2.org/job/ci_linux/7315/)

**Pull request merged**